src/lib/View/SearchListViewFilter.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace Ibexa\Platform\Search\View;
+
+use eZ\Publish\API\Repository\ContentTypeService;
+use eZ\Publish\API\Repository\SectionService;
+use eZ\Publish\Core\MVC\ConfigResolverInterface;
+use eZ\Publish\Core\MVC\Symfony\View\Event\FilterViewBuilderParametersEvent;
+use eZ\Publish\Core\MVC\Symfony\View\ViewEvents;
+use Ibexa\Platform\Bundle\SearchBundle\Form\Data\SearchData;
+use Ibexa\Platform\Bundle\SearchBundle\Form\Type\SearchType;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+
+class SearchListViewFilter implements EventSubscriberInterface
+{
+    /** @var \eZ\Publish\Core\MVC\ConfigResolverInterface */
+    private $configResolver;
+
+    /** @var \Symfony\Component\Form\FormFactoryInterface */
+    private $formFactory;
+
+    /** @var \eZ\Publish\API\Repository\SectionService */
+    private $sectionService;
+
+    /** @var \eZ\Publish\API\Repository\ContentTypeService */
+    private $contentTypeService;
+
+    public function __construct(
+        ConfigResolverInterface $configResolver,
+        FormFactoryInterface $formFactory,
+        SectionService $sectionService,
+        ContentTypeService $contentTypeService
+    ) {
+        $this->configResolver = $configResolver;
+        $this->formFactory = $formFactory;
+        $this->sectionService = $sectionService;
+        $this->contentTypeService = $contentTypeService;
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return [ViewEvents::FILTER_BUILDER_PARAMETERS => 'handleSearchForm'];
+    }
+
+    /**
+     * @throws \Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException
+     * @throws \Symfony\Component\OptionsResolver\Exception\OptionDefinitionException
+     * @throws \Symfony\Component\OptionsResolver\Exception\NoSuchOptionException
+     * @throws \Symfony\Component\OptionsResolver\Exception\MissingOptionsException
+     * @throws \Symfony\Component\OptionsResolver\Exception\AccessException
+     * @throws \Symfony\Component\OptionsResolver\Exception\InvalidOptionsException
+     * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
+     */
+    public function handleSearchForm(FilterViewBuilderParametersEvent $event): void
+    {
+        $controllerAction = $event->getParameters()->get('_controller');
+
+        if (
+            'Ibexa\Platform\Bundle\SearchBundle\Controller\SearchController::searchAction' !== $controllerAction
```

Comment:
If the controller is meant to be overridable, this won't work. If I recall correctly, we usually test on the route and not on the controller.

Reply:
That controller now returns just view from ViewBuilder, I do not see any reason to override it.
Also, I am pretty sure all ViewFilters are checked against controller and not route.

Reply:
Maybe it was the ViewBuilder, not the Filters.

But... if the controller isn't meant to be customized, what is the point of using the View API at all ? :)

Reply:
FYI:
We have internal extended search bundle, which allows admin to export search results.
So, we have controller that returns `StreamedResponse`.
Will be very helpful if controller can be customizable as well.